packages/protocol/contracts/L1/libs/LibUtils.sol

```diff
@@ -123,4 +123,38 @@ library LibUtils {
             return block.timestamp >= deadline;
         }
     }
+
+    function shouldVerifyBlocks(
+        TaikoData.Config memory _config,
+        uint64 _blockId,
+        bool _isBlockProposed
+    )
+        internal
+        pure
+        returns (bool)
+    {
+        if (_config.maxBlocksToVerify == 0) return false;
+
+        // Consider each segment of 8 blocks, verification is attempted either on block 3 if it has
+        // been
+        // proved, or on block 7 if it has been proposed. Over time, the ratio of blocks to
+        // verification attempts averages 4:1, meaning each verification attempt typically covers 4
+        // blocks. However, considering worst cases caused by blocks being proved out of order, some
+        // verification attempts may verify few or no blocks. In such cases, additional
+        // verifications are needed to catch up. Consequently, the `maxBlocksToVerify` parameter
+        // should be set high enough to allow for efficient catch-up.
+
+        // Now lets use `maxBlocksToVerify` as an input to calculate the size of each block
+        // segment, instead of using 8 as a constant.
+        uint256 segmentSize = _config.maxBlocksToVerify / 2;
+
+        if (segmentSize <= 1) return true;
+
+        uint256 remainder;
+        unchecked {
+            remainder = _isBlockProposed ? segmentSize - 1 : segmentSize / 2 - 1;
+        }
+
+        return _blockId % segmentSize == remainder;
```

Comment:
just `_blockId % segmentSize == 0;` not the same?

Reply:
Statistically yes, but we don't want a block's proposer AND prove to pay for verifications.